tests/suite/math/call.typ

```diff
@@ -0,0 +1,95 @@
+// Test math function call edge cases.
+
+// Note: 2d argument calls are tested for matrices in `mat.typ`
+
+--- math-call-non-func ---
+$ pi(a) $
+$ pi(a,) $
+$ pi(a,b) $
+$ pi(a,b,) $
+
+--- math-call-repr ---
+#let args(..body) = body
+#let check(it, r) = test-repr(it.body.text, r)
+#check($args(a)$, "([a])")
+#check($args(a,)$, "([a])")
+#check($args(a,b)$, "([a], [b])")
+#check($args(a,b,)$, "([a], [b])")
+#check($args(,a,b,,,)$, "([], [a], [b], [], [])")
+
+--- math-call-2d-non-func ---
+// Error: 6-7 expected content, found array
+// Error: 8-9 expected content, found array
+$ pi(a;b) $
+
+--- math-call-2d-semicolon-priority ---
+$ mat("math" ; "wins") $
+$ mat(#"code"; "wins") $
```

Comment:
This is testing how semicolons in code mode end statements, and are greedily consumed by the code expression parser so long as there's no space to separate them. (I also forgot to add a hash tag to the string for math mode, so that the code is showing how the semicolon is not consumed when there's a space after the string).

Reply:
oh right, I was quite confused but now I see it